### PR TITLE
Preparations to move the user guide to neos.io

### PR DESCRIPTION
**Pull request by beheist**
Reasoning: See https://discuss.neos.io/t/refactoring-the-neos-user-guide/3256/2 with @rolandschuetz.

Will introduce a new document type for documentation pages and some features / node types which work there only.

!!! ATTN !!! We are enabling content dimensions here, so the according node migration needs to be run after this is taken live.

### CODE REVIEW

**Comment by beheist** [comment]
Ok, we're ready here.

File: Packages/Sites/Neos.NeosIo/Configuration/NodeTypes.DocumentationPage.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+###########################################
+# Documentation page, where special node types
+# are available.
+###########################################
+
+'Neos.NeosIo:DocumentationPage':
+  superTypes:
+    'Neos.NodeTypes:Page': true
+#    'Neos.NeosIo:KeyVisualMixin': true
```
[the remark this answers is not in the document]
Comment (rolandschuetz) [reply Packages/Sites/Neos.NeosIo/Configuration/NodeTypes.DocumentationPage.yaml]: I will take care of this after the initial documentation content is added, I'm not sure yet how it best works. So I would like to experiment a bit.

File: Packages/Sites/Neos.NeosIo/Configuration/NodeTypes.DocumentationPage.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  childNodes:
+    main:
+      type: 'Neos.Neos:ContentCollection'
```
[the remark this answers is not in the document]
Comment (rolandschuetz) [reply Packages/Sites/Neos.NeosIo/Configuration/NodeTypes.DocumentationPage.yaml]: I will take care of this after the initial documentation content is added, I'm not sure yet how it best works. So I would like to experiment a bit.

File: Packages/Sites/Neos.NeosIo/Resources/Private/Templates/FusionObjects/DocumentationPageSubMenu.html [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+<ul{attributes -> f:format.raw()}>
+	<f:render section="itemsList" arguments="{items: items}" />
+</ul>
+
+<f:section name="itemsList">
+	<f:for each="{items}" as="item">
+		<li{fusion:render(path: '{item.state}.attributes', context: {item: item}) -> f:format.raw()}>
+		<neos:link.node node="{item.node}" section="{item.anchorName}">{item.label}</neos:link.node>
+		</li>
+		<f:if condition="{item.subItems}">
```
[the remark this answers is not in the document]
Comment (rolandschuetz) [reply Packages/Sites/Neos.NeosIo/Resources/Private/Templates/FusionObjects/DocumentationPageSubMenu.html]: Semantically you are right. But the way the CSS currently works, it is visually correct. I think it's not worth the effort to adopt the CSS.

File: Packages/Sites/Neos.NeosIo/Resources/Private/Fusion/FusionObjects/DocumentationPageSubMenu.fusion [hunk 4]
```diff
@@ -0,0 +1,52 @@
+prototype(Neos.NeosIo:DocumentationPageSubMenu) < prototype(Neos.Neos:Menu) {
+    @class = 'Neos\\NeosIo\\Fusion\\DocumentationPageSubMenuImplementation'
+    templatePath = 'resource://Neos.NeosIo/Private/Templates/FusionObjects/DocumentationPageSubMenu.html'
+
+    itemCollection = Neos.Fusion:RawCollection {
+        collection = ${q(site).find('[instanceof Neos.NeosIo:DocumentationPage]')}
```
[the remark this answers is not in the document]
Comment (rolandschuetz) [reply Packages/Sites/Neos.NeosIo/Resources/Private/Fusion/FusionObjects/DocumentationPageSubMenu.fusion]: Yes, you are right. That's confusion. Bastian is busy, so I will provide a fix for that.

**Comment by rolandschuetz** [comment]
@Sebobo @beheist I have fixed the above-discussed issues. Please review :)

**Comment by rolandschuetz** [comment]
Closed in favor of #219

**Pull request closed without merging**